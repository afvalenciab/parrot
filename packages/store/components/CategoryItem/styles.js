import { makeStyles } from '@material-ui/core/styles';

export const useStyles = makeStyles(theme => ({
  card: {
    display: 'flex',
    flexShrink: '0',
    width: '216px',
    border: '1px solid transparent',
  },
  active: {
    border: `1px solid ${theme.palette.parrot}`,
    backgroundColor: 'rgba(240, 78,74,.07)',
  },
  cardContent: {
    paddingBottom: '0.8rem !important',
    padding: '0.8rem',
  },
  title: {
    fontWeight: 'bold',
    lineHeight: 1,
    paddingBottom: '1rem',
  },
  items: {
    color: theme.palette.grey2,
  },
}));
