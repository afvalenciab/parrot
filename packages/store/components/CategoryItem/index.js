import { Card, CardActionArea, CardContent, Typography } from '@material-ui/core';
import { useTheme } from '@material-ui/core/styles';
import clsx from 'clsx';

import { useStyles } from './styles';

export default function CategoryItem({ category, onSelectCategory, isActive }) {
  const theme = useTheme();
  const classes = useStyles(theme);

  return (
    <Card
      className={clsx(classes.card, { [classes.active]: isActive })}
      data-testid={`category-${category.uuid}`}>
      <CardActionArea onClick={onSelectCategory} data-testid={`btn-${category.uuid}`}>
        <CardContent className={classes.cardContent}>
          <Typography variant="subtitle1" component="h3" className={classes.title}>
            {category.name}
          </Typography>

          <Typography variant="body2" component="p" className={classes.items}>
            {`${category.totalProducts} ${category.totalProducts === 1 ? 'artículo' : 'artículos'}`}
          </Typography>
        </CardContent>
      </CardActionArea>
    </Card>
  );
}
