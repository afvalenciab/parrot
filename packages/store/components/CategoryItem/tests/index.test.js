import React from 'react';
import { render, fireEvent } from '@testing-library/react';

import CategoryItem from 'components/CategoryItem';

describe('<CategoryItem />', () => {
  const props = {
    category: { uuid: '123', name: 'Category 1', totalProducts: '2' },
    onSelectCategory: jest.fn(),
    isActive: true,
  };

  it('Should render correctly', () => {
    const { container } = render(<CategoryItem {...props} />);

    expect(container).toMatchSnapshot();
  });

  it('Should render the category-123 when the component is mounted', () => {
    const { getByTestId } = render(<CategoryItem {...props} />);

    const category = getByTestId('category-123');

    expect(category).toBeInTheDocument();
  });

  it('Should call the function onSelectCategory when the category-123 is selected', () => {
    const { getByTestId } = render(<CategoryItem {...props} />);

    const btnCategory = getByTestId('btn-123');

    fireEvent.click(btnCategory);

    expect(props.onSelectCategory).toHaveBeenCalled();
  });
});
