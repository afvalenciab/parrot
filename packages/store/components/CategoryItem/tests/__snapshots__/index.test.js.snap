// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<CategoryItem /> Should render correctly 1`] = `
<div>
  <div
    class="MuiPaper-root MuiCard-root makeStyles-card-1 makeStyles-active-2 MuiPaper-elevation1 MuiPaper-rounded"
    data-testid="category-123"
  >
    <button
      class="MuiButtonBase-root MuiCardActionArea-root"
      data-testid="btn-123"
      tabindex="0"
      type="button"
    >
      <div
        class="MuiCardContent-root makeStyles-cardContent-3"
      >
        <h3
          class="MuiTypography-root makeStyles-title-4 MuiTypography-subtitle1"
        >
          Category 1
        </h3>
        <p
          class="MuiTypography-root makeStyles-items-5 MuiTypography-body2"
        >
          2 art√≠culos
        </p>
      </div>
      <span
        class="MuiCardActionArea-focusHighlight"
      />
      <span
        class="MuiTouchRipple-root"
      />
    </button>
  </div>
</div>
`;
